Source/WebInspectorUI/UserInterface/Views/ScopeBar.js

```diff
@@ -148,20 +154,23 @@ WI.ScopeBar = class ScopeBar extends WI.NavigationItem
 
     _itemSelectionDidChange(event)
     {
+        if (this._ignoreItemSelectedEvent)
+            return;
+
+        this._ignoreItemSelectedEvent = true;
+
         var sender = event.target;
         var item;
 
-        // An exclusive item was selected, unselect everything else.
         if (sender.exclusive && sender.selected) {
-            for (var i = 0; i < this._items.length; ++i) {
-                item = this._items[i];
+            // An exclusive item was selected, deselect everything else.
+            for (var item of this._ungroupedItems) {
```

Comment:
rather than keep a second list in `_ungroupedItems` i think we could just do this
```js
for (let item of this._items) {
    if (this._shouldGroupNonExclusiveItems && !item.exclusive)
        continue;

    if (item !== sender)
        item.selected = false;
}

if (this._multipleItem && this._multipleItem !== sender)
    this._multipleItem.selected = false;
```